src/axolotl/utils/callbacks/vessl_.py

```diff
@@ -0,0 +1,27 @@
+"""Vessl module for trainer callbacks"""
+import logging
+from typing import Dict
+
+import vessl
+from transformers import TrainerCallback, TrainerControl, TrainerState
+from transformers.training_args import TrainingArguments
+
+LOG = logging.getLogger("axolotl.callbacks")
+
+
+class VesslLogMetricsCallback(TrainerCallback):
+    """Callback to send training metrics to VESSL AI"""
+
+    def __init__(self, credential_path: str) -> None:
+        vessl.configure(credentials_file=credential_path)
+
+    def on_log(
+        self,
+        args: TrainingArguments,  # pylint: disable=unused-argument
+        state: TrainerState,
+        control: TrainerControl,  # pylint: disable=unused-argument
+        logs: Dict[str, float],
+        **kwargs  # pylint: disable=unused-argument
+    ):
+        if state.is_world_process_zero:
```

Comment:
can you explain where you copied this code from, please?
I am wondering why it is different to the following
https://github.com/vessl-ai/examples/blob/ebeae1c430509d619c380c56923c645cbd02f610/llama-factory/src/llmtuner/extras/callbacks.py#L162-L187

Reply:
I take it from wandb integration:
https://github.com/huggingface/transformers/blob/main/src/transformers/integrations/integration_utils.py#L824

The difference between `world_process_zero` and `local_process_zero` is explained here:
https://github.com/huggingface/transformers/blob/main/src/transformers/trainer_callback.py#L77